src/gui/editors/TrackContainerView.cpp

```diff
@@ -349,6 +356,28 @@ void TrackContainerView::setPixelsPerBar( int ppb )
 	}
 }
 
+void TrackContainerView::setVerticalScale(double h)
+{
+	for (const auto& trackView : m_trackViews)
+	{
+		int new_height = lround(h * DEFAULT_TRACK_HEIGHT);
+		int cur_height = trackView->getTrack()->getHeight();
+
+		// If a track has been manually resized, calculate the new height by scaling the old height rather than just setting it.
+		if(cur_height != lround((m_trackHeightScale * DEFAULT_TRACK_HEIGHT)))
+			new_height = lround(cur_height * (h / m_trackHeightScale));
+
+		// clamp the value to prevent it from becoming smaller than the zoom level.
+		if(new_height < lround(h * DEFAULT_TRACK_HEIGHT))
+			new_height = lround(h * DEFAULT_TRACK_HEIGHT);
```

Comment:
```suggestion
		if (new_height < lround(h * DEFAULT_TRACK_HEIGHT))
		{
			new_height = lround(h * DEFAULT_TRACK_HEIGHT);
		}
```

Reply:
And here as well, `std::lround` could be used